__tests__/app.test.js

```diff
@@ -309,48 +313,77 @@ describe.only("PATCH - /api/reviews/:review_id", () => {
         });
       });
   });
-  test("GET - 404: responds with an error message, when passed an invalid review_id", () => {
-    return request(app)
-      .patch("/api/reviews/999")
-      .send({
-        inc_votes: -100,
-      })
-      .expect(404)
-      .then(({ body }) => {
-        expect(body.msg).toBe("Comments with review id 999 not found!");
-      });
-  });
-  test("GET - 400: responds with an error message, when passed an invalid review id", () => {
-    return request(app)
-    .patch("/api/reviews/nonsense")
-    .send({
-      inc_votes: -100,
-    })
-      .expect(400)
-      .then(({ body }) => {
-        expect(body.msg).toBe("invalid query!");
-      });
+  describe("-error tests", () => {
+    test("GET - 404: responds with an error message, when passed an invalid review_id", () => {
+      return request(app)
+        .patch("/api/reviews/999")
+        .send({
+          inc_votes: -100,
+        })
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Comments with review id 999 not found!");
+        });
+    });
+    test("GET - 400: responds with an error message, when passed an invalid review id", () => {
+      return request(app)
+        .patch("/api/reviews/nonsense")
+        .send({
+          inc_votes: -100,
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("invalid query!");
+        });
+    });
+    test("GET 400: responds with update failed message if passed an invalid increment", () => {
+      return request(app)
+        .patch("/api/reviews/4")
+        .send({
+          inc_votes: "one hundred",
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("update failed, increment data type incorrect");
+        });
+    });
+    test("GET 400: responds with invalid request msg if inc_votes spelt incorrectly", () => {
+      return request(app)
+        .patch("/api/reviews/4")
+        .send({
+          inckkk_votez: 100,
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("invalid request");
+        });
+    });
   });
-  test("GET 400: responds with update failed message if passed an invalid increment", () => {
+});
+
+describe("GET - /api/users", () => {
+  test("GET - 200: responds with an array of user objects ", () => {
     return request(app)
-    .patch("/api/reviews/4")
-    .send({
-      inc_votes: "one hundred",
-    })
-      .expect(400)
+      .get("/api/users")
+      .expect(200)
       .then(({ body }) => {
-        expect(body.msg).toBe("update failed, increment data type incorrect");
+        body.users.forEach((user) => {
+          expect.objectContaining({
+            username: expect.any(String),
+            name: expect.any(String),
+            avatar_url: expect.any(String),
+          });
```

Comment:
if we are using a forEach like this we need an additional expect statement checking that the length of array is greater than 0